### PR TITLE
feat(schema): Add workspace invitation schema

### LINKED ISSUE #552: SCHEMA: Add "Get All Workspace Invitations" request and response types
## Description

In the `schema` package, add the schemas for the following types:
- `GetWorkspaceInvitationsRequest`
- `GetWorkspaceInvitationsResponse`

## Solution

- Add the schemas in [here](https://github.com/keyshade-xyz/keyshade/blob/develop/packages/schema/src/workspace/index.ts)
- Add the inferred types in [here](https://github.com/keyshade-xyz/keyshade/blob/develop/packages/schema/src/workspace/index.types.ts)
- Add relevant tests in [here](https://github.com/keyshade-xyz/keyshade/blob/develop/packages/schema/tests/workspace.spec.ts)

## Additional context

Depends on #550

### PR DESCRIPTION
**Pull request by Souvik9205**
## Description

**Schema and relevant tests for Workspace's invitation lists**  <br />

Fixes #552

___

### **PR Type**
Enhancement, Tests

___

### **Description**
- Added schemas for `GetInvitationRequest` and `GetInvitationResponse`.

- Defined types for the new schemas in `index.types.ts`.

- Implemented validation tests for the new schemas.

- Enhanced workspace schema functionality with invitation-related features.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Added schemas for workspace invitation handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/src/workspace/index.ts

<li>Added <code>GetInvitationRequestSchema</code> and <code>GetInvitationResponseSchema</code>.<br> <li> Defined <code>WorkspaceInvitationDetailsSchema</code> for invitation details.<br> <li> Integrated schemas with existing pagination structures.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/612/files#diff-f652e357283c083dd481b06b151cd327ef1fde48bc60c6bb86dc0d4f9dc185f0">+24/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>index.types.ts</strong><dd><code>Defined types for new invitation schemas</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/src/workspace/index.types.ts

<li>Added types for <code>GetInvitationRequestSchema</code> and <br><code>GetInvitationResponseSchema</code>.<br> <li> Extended type definitions for workspace invitation handling.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/612/files#diff-2be248985491f04e9ad2a0b63db93287f03bd8ef937aca57bc4c2f3205d2ec75">+7/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>workspace.spec.ts</strong><dd><code>Added tests for workspace invitation schemas</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/schema/tests/workspace.spec.ts

<li>Added validation tests for <code>GetInvitationRequestSchema</code>.<br> <li> Added validation tests for <code>GetInvitationResponseSchema</code>.<br> <li> Included edge case tests for invalid inputs.

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/612/files#diff-cc5087f98800be588f42b5237f6e2c07fed0f17e812415ddc8bda7385ca235da">+68/-1</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by Souvik9205** [comment]
@rajdip-b please review `WorkspaceInvitationDetailsSchema` and if there is any more type error can you please suggest that? I already went through [https://github.com/keyshade-xyz/keyshade/pull/586/files#diff-a92613f23cf2834d52eaedb87c4e8c460d48b6fca8df4ccc968b2a8af19a9c3fR432-R450](here)

File: packages/schema/src/workspace/index.ts [hunk 1]
```diff
@@ -149,3 +149,27 @@ export const GlobalSearchResponseSchema = z.object({
     })
   )
 })
+
+export const GetInvitationRequestSchema = PageRequestSchema
+
+const WorkspaceInvitationDetailsSchema = z.object({
```
Comment (rajdip-b) [inline packages/schema/src/workspace/index.ts:155]: We don't need this to be separate. You can directly put this into `GetWorkspaceInvitationsResponse`

File: packages/schema/src/workspace/index.ts
Comment (rajdip-b) [inline packages/schema/src/workspace/index.ts:1]: The schema names should be `GetWorkspaceInvitationsRequest` and `GetWorkspaceInvitationsResponse`

**Review by rajdip-b — CHANGES_REQUESTED** [review]

**Review by rajdip-b — APPROVED** [review]

File: packages/schema/src/workspace/index.types.ts [hunk 2]
```diff
@@ -66,3 +68,11 @@ export type ExportDataResponse = z.infer<typeof ExportDataResponseSchema>
 export type GlobalSearchRequest = z.infer<typeof GlobalSearchRequestSchema>
 
 export type GlobalSearchResponse = z.infer<typeof GlobalSearchResponseSchema>
+
+export type GetInvitationRequest = z.infer<
```
Comment (rajdip-b) [inline packages/schema/src/workspace/index.types.ts:72]: You missed renaming these :)

**Review by rajdip-b — CHANGES_REQUESTED** [review]

**Review by rajdip-b — APPROVED** [review]

**Comment by rajdip-b** [comment]
@Souvik9205 could you please ping me over discord? Here's my ID: _the_heisenberg.
